controllers/capabilities/backend_controller.go

```diff
@@ -74,12 +76,57 @@ func (r *BackendReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		reqLogger.V(1).Info(string(jsonData))
 	}
 
+	// Ignore deleted Backends, this can happen when foregroundDeletion is enabled
+	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
+	if backend.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(backend, backendFinalizer) {
+		err = controllerhelper.ReconcileFinalizers(backend, r.Client(), backendFinalizer)
```

Comment:
First delete backend from 3scale, if that works, remove the finalizer. 

Have always in mind that the implementation needs to be idempotent and you can expect many reconciliation loop calls to delete the backend.

Reply:
good catch. Thank you